vhdbuilder/packer/vhd-scanning.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+set -eux
+
+TRIVY_SCRIPT_PATH="trivy-scan.sh"
+TEST_RESOURCE_PREFIX="vhd-scanning"
+VM_NAME="$TEST_RESOURCE_PREFIX-vm"
+VHD_IMAGE="$MANAGED_SIG_ID"
+SIG_CONTAINER_NAME="vhd-scans"
+TEST_VM_ADMIN_USERNAME="azureuser"
+STORAGE_ACCOUNT_NAME="vhdbuildereastustest"
+
+set +x
+TEST_VM_ADMIN_PASSWORD="TestVM@$(date +%s)"
+set -x
+
+
+RESOURCE_GROUP_NAME="$TEST_RESOURCE_PREFIX-$(date +%s)-$RANDOM"
+az group create --name $RESOURCE_GROUP_NAME --location ${AZURE_LOCATION} --tags 'source=AgentBaker'
+
+if [ "$OS_VERSION" == "18.04" ]; then
+    echo "Skipping scanning for 18.04"
+    exit 0
+fi
+
+function cleanup() {
+    echo "Deleting resource group ${RESOURCE_GROUP_NAME}"
+    az group delete --name $RESOURCE_GROUP_NAME --yes --no-wait
+}
+trap cleanup EXIT
+
+VM_OPTIONS="--size Standard_DS1_v2"
+if [[ "${ARCHITECTURE,,}" == "arm64" ]]; then
+    VM_OPTIONS="--size Standard_D2pds_v5"
+elif [[ "${FEATURE_FLAGS,,}" == "kata" ]]; then
+    VM_OPTIONS="--size Standard_D4ds_v5"
+fi
+
+if [[ "${OS_TYPE}" == "Linux" && "${ENABLE_TRUSTED_LAUNCH}" == "True" ]]; then
+    VM_OPTIONS+="--security-type TrustedLaunch --enable-secure-boot true --enable-vtpm true"
+fi
+
+#fix identity string
+az vm create --resource-group $RESOURCE_GROUP_NAME \
+    --name $VM_NAME \
+    --image $VHD_IMAGE \
+    --admin-username $TEST_VM_ADMIN_USERNAME \
+    --admin-password $TEST_VM_ADMIN_PASSWORD \
+    --os-disk-size-gb 50 \
+    --assign-identity "/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/vhd-scanning-UAMI" \
+    ${VM_OPTIONS}
+
+FULL_PATH=$(realpath $0)
+CDIR=$(dirname $FULL_PATH)
+SCRIPT_PATH="$CDIR/$TRIVY_SCRIPT_PATH"
+az vm run-command invoke \
+    --command-id RunShellScript \
+    --name $VM_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --scripts @$SCRIPT_PATH
+
+if [[ "$OS_SKU" = "Ubuntu" && "$SIG_IMAGE_NAME" == *2004* ]]; then
```

Comment:
could we look at turning all those `az vm run-command invoke` into a single shell scripts ?

```
   az vm run-command invoke \
         --command-id RunShellScript \
         --name $VM_NAME \
         --resource-group $RESOURCE_GROUP_NAME \
         --scripts @$UPLOAD_SCANNING_RESULTS_SCRIPT_PATH
```